lib/serverspec/type/hadoop_config.rb

```diff
@@ -0,0 +1,28 @@
+module Serverspec::Type
+  class HadoopConfig < Base
+    def initialize(name=nil)
+      super
+
+      begin
+        require 'nokogiri'
+      rescue LoadError
+        fail "nokogiri is not available. Try installing it."
+      end
+    end
+
+    def value
+      regx = /#{@name}/
```

Comment:
If you intend to reduce regular expression compile time, it's OK. But `regx` is not common. `regex` is common.